### ISSUE TITLE
kdominos: dumps core on human win

### ISSUE THREAD

**Issue by BartMassey** [keith-packard/kgames#8]
Placed the winning domino to empty my rack. Game immediately dumped core. Was repeatable. Have the executable and core file in case more debugging is needed. Stack trace:

```
(gdb) where
#0  0x000055ecafb32ba2 in DominoWidth (w=0x55ecafcac960, d=0x0)
    at ../dominos/Dominos.c:85
#1  0x000055ecafb32fba in BoardSize
    (w=0x55ecafcac960, b=0x0, r=0x7fffaa9502d0, x=0, y=0)
    at ../dominos/Dominos.c:212
#2  0x000055ecafb33176 in PreferredSize
    (w=0x55ecafcac960, width=0x7fffaa9503d8, height=0x7fffaa9503da, x=0x0, y=0x0) at ../dominos/Dominos.c:240
#3  0x000055ecafb34759 in QueryGeometry
    (gw=0x55ecafcac960, intended=0x7fffaa9503f0, preferred=0x7fffaa9503d0)
    at ../dominos/Dominos.c:659
#4  0x00007f6bf67a0235 in XtQueryGeometry ()
    at /usr/lib/x86_64-linux-gnu/libXt.so.6
#5  0x00007f6bf6bc68ea in LayoutChild
    (pw=0x55ecafcac650, child=0x55ecafcac960, geomp=0x0, xp=0x7fffaa95045e, yp=0x7fffaa95045c, widthp=0x7fffaa95045a, heightp=0x7fffaa950458)
    at ../Xkw/KPorthole.c:76
#6  0x00007f6bf6bc6b7c in KPortholeResize (gw=0x55ecafcac650)
    at ../Xkw/KPorthole.c:159
#7  0x00007f6bf679fe86 in XtConfigureWidget ()
    at /usr/lib/x86_64-linux-gnu/libXt.so.6
#8  0x00007f6bf6bc0f00 in SetSizes (box=0x55ecafc3e000, x=0, y=76)
    at ../Xkw/Layout.c:882
--Type <RET> for more, q to quit, c to continue without paging--c
#9  0x00007f6bf6bc0f3b in SetSizes (box=0x55ecafc3e6a0, x=0, y=76) at ../Xkw/Layout.c:892
#10 0x00007f6bf6bc12c4 in LayoutLayout (l=0x55ecafc1b160, attemptResize=1) at ../Xkw/Layout.c:990
#11 0x00007f6bf6bbe47d in GeometryManager (child=0x55ecafc3e8c0, request=0x7fffaa950790, reply=0x7fffaa9507b0) at ../Xkw/Layout.c:177
#12 0x00007f6bf679f39d in _XtMakeGeometryRequest () at /usr/lib/x86_64-linux-gnu/libXt.so.6
#13 0x00007f6bf679f969 in XtMakeResizeRequest () at /usr/lib/x86_64-linux-gnu/libXt.so.6
#14 0x00007f6bf6bc122c in LayoutLayout (l=0x55ecafc3e8c0, attemptResize=1) at ../Xkw/Layout.c:976
#15 0x00007f6bf6bbe47d in GeometryManager (child=0x55ecafcab740, request=0x7fffaa950a20, reply=0x7fffaa950a40) at ../Xkw/Layout.c:177
#16 0x00007f6bf679f39d in _XtMakeGeometryRequest () at /usr/lib/x86_64-linux-gnu/libXt.so.6
#17 0x00007f6bf67b4333 in XtSetValues () at /usr/lib/x86_64-linux-gnu/libXt.so.6
#18 0x00007f6bf6bc1848 in MessageEnd (w=0x55ecafcab740) at ../Xkw/Message.c:165
#19 0x00007f6bf6bc1925 in MessageV (w=0x55ecafcab740, format=0x55ecafb36b19 "%s have %d point%s.", args=0x7fffaa9512d8) at ../Xkw/Message.c:183
#20 0x00007f6bf6bc19ce in Message (w=0x55ecafcab740, format=0x55ecafb36b19 "%s have %d point%s.") at ../Xkw/Message.c:192
#21 0x000055ecafb30ae0 in DisplayScores () at ../dominos/dominos.c:253
#22 0x000055ecafb30c98 in Score () at ../dominos/dominos.c:293
#23 0x000055ecafb30d81 in GameOver () at ../dominos/dominos.c:329
#24 0x000055ecafb307d9 in MakeMove (player=0x55ecafb3bbe0 <player>, source=0x55ecafcb0350, target=0x55ecafcb0b00, dir=East, orientation=North) at ../dominos/dominos.c:147
#25 0x000055ecafb30898 in PlayEdgeFunc (source=0x55ecafcb0350, target=0x55ecafcb0b00, dir=East, orientation=North, data=0x7fffaa9517f0) at ../dominos/dominos.c:176
#26 0x000055ecafb34da6 in ForEachEdge (d=0x55ecafcb0b00, dir=East, f=0x7fffaa9517a0) at ../dominos/find.c:41
#27 0x000055ecafb34eb0 in FindEdge (d=0x55ecafcb0b00, search_dir=East, f=0x7fffaa9517a0) at ../dominos/find.c:73
#28 0x000055ecafb34e99 in FindEdge (d=0x55ecafcb0380, search_dir=East, f=0x7fffaa9517a0) at ../dominos/find.c:72
#29 0x000055ecafb34e99 in FindEdge (d=0x55ecafcb0bf0, search_dir=East, f=0x7fffaa9517a0) at ../dominos/find.c:72
#30 0x000055ecafb34e99 in FindEdge (d=0x55ecafcb0bc0, search_dir=East, f=0x7fffaa9517a0) at ../dominos/find.c:72
#31 0x000055ecafb34e99 in FindEdge (d=0x55ecafcb0c50, search_dir=East, f=0x7fffaa9517a0) at ../dominos/find.c:72
#32 0x000055ecafb34e99 in FindEdge (d=0x55ecafcb0b30, search_dir=East, f=0x7fffaa9517a0) at ../dominos/find.c:72
#33 0x000055ecafb34e2f in FindEdge (d=0x55ecafcb0c20, search_dir=East, f=0x7fffaa9517a0) at ../dominos/find.c:58
#34 0x000055ecafb34e99 in FindEdge (d=0x55ecafcb0b60, search_dir=East, f=0x7fffaa9517a0) at ../dominos/find.c:72
#35 0x000055ecafb34e99 in FindEdge (d=0x55ecafcb0320, search_dir=East, f=0x7fffaa9517a0) at ../dominos/find.c:72
#36 0x000055ecafb34e2f in FindEdge (d=0x55ecafcb0d10, search_dir=North, f=0x7fffaa9517a0) at ../dominos/find.c:58
#37 0x000055ecafb34f2f in FindPlays (board=0x55ecafcb0d10, domino=0x55ecafcb0350, func=0x55ecafb30823 <PlayEdgeFunc>, data=0x7fffaa9517f0) at ../dominos/find.c:94
#38 0x000055ecafb30940 in Play (player=0x55ecafb3bbe0 <player>, domino=0x55ecafcb0350, target=0x55ecafcb0b00, dir=East, dist=69) at ../dominos/dominos.c:205
#39 0x000055ecafb3208a in BoardCallback (w=0x55ecafcac1f0, closure=0x0, data=0x7fffaa9518a0) at ../dominos/dominos.c:879
#40 0x000055ecafb338ce in ActionSelect (gw=0x55ecafcac1f0, event=0x7fffaa951df0, params=0x55ecafcac460, num_params=0x55ecafcac420) at ../dominos/Dominos.c:418
#41 0x00007f6bf67c323f in  () at /usr/lib/x86_64-linux-gnu/libXt.so.6
#42 0x00007f6bf67c3ce3 in  () at /usr/lib/x86_64-linux-gnu/libXt.so.6
#43 0x00007f6bf67c4986 in _XtTranslateEvent () at /usr/lib/x86_64-linux-gnu/libXt.so.6
#44 0x00007f6bf679c4ae in XtDispatchEventToWidget () at /usr/lib/x86_64-linux-gnu/libXt.so.6
#45 0x00007f6bf679cee7 in  () at /usr/lib/x86_64-linux-gnu/libXt.so.6
#46 0x00007f6bf679d053 in XtDispatchEvent () at /usr/lib/x86_64-linux-gnu/libXt.so.6
#47 0x00007f6bf67a8be0 in XtAppProcessEvent () at /usr/lib/x86_64-linux-gnu/libXt.so.6
#48 0x00007f6bf679d463 in XtAppMainLoop () at /usr/lib/x86_64-linux-gnu/libXt.so.6
#49 0x000055ecafb32b58 in main (argc=1, argv=0x7fffaa952068) at ../dominos/dominos.c:1196
```

**Comment by keith-packard** [comment]
I think I fixed this in 359340ee1b21e3e351d6cea0840f0ca5242d829b

**Comment by BartMassey** [comment]
Verified fixed.
